src/org/labkey/test/components/glassLibrary/heatmap/HeatMap.java

```diff
@@ -0,0 +1,224 @@
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static org.labkey.test.WebDriverWrapper.WAIT_FOR_JAVASCRIPT;
+import static org.labkey.test.util.TestLogger.log;
+
+/**
+ * Manipulates the HeatMap component from labkey-ui-components (components/src/internal/components/heatmap/HeatMap.tsx)
+ */
+public class HeatMap extends WebDriverComponent<HeatMap.ElementCache>
+{
+    private final WebElement _el;
+    private final WebDriver _driver;
+    private final Locator.XPathLocator _rowsLoc = Locator.tag("tbody").childTag("tr").withoutClass("grid-empty").withoutClass("grid-loading");
+    private final Locator _loadingRowLoc = Locator.css("tbody tr.grid-loading");
+    private final Locator _spinnerLoc = Locator.css("span i.fa-spinner");
+    private final Locator _emptyRowLoc = Locator.css("tbody tr.grid-empty");
+
+    protected HeatMap(WebElement element, WebDriver driver)
+    {
+        _el = element;
+        _driver = driver;
+    }
+
+    @Override
+    public WebElement getComponentElement()
+    {
+        return _el;
+    }
+
+    @Override
+    public WebDriver getDriver()
+    {
+        return _driver;
+    }
+
+    public String getColumnByIndex(int index)
+    {
+        return getColumnNames().get(index);
+    }
+
+    public List<String> getColumnNames()
+    {
+        List<WebElement> headerCells = elementCache().headerCells();
+        return getWrapper().getTexts(headerCells);
+    }
+
+    /**
+     * Each row has an anchor tag in its first column; usually it's a link to a sampletype or an assay.
+     * We will refer to that link as the row's name
+     * @return
+     */
+    public List<String> getRowNames()
+    {
+        List<String> rowNames = new ArrayList<>();
+        for (WebElement row : getRows())
+        {
+            WebElement entityLink = Locator.xpath("//td/div/a").findElement(row);
+            rowNames.add(entityLink.getText());
+        }
+        return rowNames;
+    }
+
+    public Optional<WebElement> getOptionalRow(String linkText)
+    {
+        List<WebElement> rows = getRows();
+        Locator matchingLink = Locator.tag("td").append(Locator.linkWithText(linkText));
+
+        for (WebElement row : rows)
+        {
+            if (matchingLink.existsIn(row))
+            {
+                return Optional.of(row);
+            }
+        }
+
+        log("No rows matching ["+matchingLink+"] were found");
+        return Optional.empty();
+    }
+
+    public WebElement getEntityLink(String linkText)
+    {
+        WebElement row = getOptionalRow(linkText).get();
+        return Locator.linkWithText(linkText).waitForElement(row, WAIT_FOR_JAVASCRIPT);
+    }
+
+    public WebElement getSummaryLink(String rowLinkText)
```

Comment:
The parameter names should be more consistent. Here it is 'rowLinkText', for getOptionalRow the same value is called 'linkText' and in another method these values are returned as rowNames (getRowNames).

Since there is the function getRowNames I would go with rowNames for consistency.